### ISSUE TITLE
How do we make sure we can release new versions?

### ISSUE THREAD

**Issue by simeg** [guyht/vimari#100]
Hey everyone,

So as you might've seen in the README we are currently blocked from releasing new versions because no one maintaining this project has an Apple developer subscription. They are $100 a year and no one of the maintainers has any other use for such a subscription, so it makes no sense to purchase such subscription for this reason only.

After adding that info to the README I got a few people contacting me about wanting to help (big thanks!), but between the time I added that info and now Apple changed how they handle Safari extensions. Before you could just compile your extension in Safari, get a `.safariextz` and use that binary to share your extension. No person dependency. Now it seems like you must publish your extension to the Safari gallery (see #99).

This makes it really hard because someone will have to _own_ this extension in the gallery, and having a person dependency in an open-source world is never a good idea. So even though one of these fine gentlemen who contacted me during the past days would submit a new version of the extension we would be tied to their account for new versions. Not ideal.

Me and @jad talked via e-mail and he proposed crowd funding a subscription. That got me thinking we could maybe have a special account that multiple people have access to just for this purpose (if that's allowed). I think this is a good idea, and I don't think $100 would take too long to collect.

I would appreciate your opinions on this, and if people think crowd funding a subscription is a good idea I will set it up.

@guyht @jad

**Comment by jad** [comment]
> So even though one of these fine gentlemen who contacted me during the past days would submit a new version of the extension we would be tied to their account for new versions. Not ideal.

It's true that it's not ideal, though IIRC apps can be transferred between accounts and I **think** the bundle ID can stay the same (should be confirmed though).

Also, multiple people can have access to a single developer account and multiple people can publish apps through that account, so it would never be gated on one person anyway just to get a new version out.

> Me and @jad talked via e-mail and he proposed crowd funding a subscription.

I don't think I ever proposed crowd funding FWIW. I have an account I pay for every year anyway, and I'm a Vimari user and I want it to work in Safari 12, so I'm fine just publishing the extension under my account. I understand the other legitimate concerns about ownership, but I just want to be clear that I'm willing to do this on my own without collecting money from anyone.

> That got me thinking we could maybe have a special account that multiple people have access to just for this purpose (if that's allowed).

I think this would be fine if you wanted to try it. It's not that much different than the previous setup though, because the account is still ultimately owned by one person.

Another thing to consider is that if people contribute money, what, if any, obligations should be upheld in return. For example, one could imagine a commitment to keep the extension published and updated to support new operating system versions over the course of a year at a minimum. Maybe critical bug fixes too. Then you'd have to think about if it also includes any kind of active feature development. Whatever the decision is, that should be made clear when asking for monetary contributions.

**Comment by simeg** [comment]
>Also, multiple people can have access to a single developer account and multiple people can publish apps through that account, so it would never be gated on one person anyway just to get a new version out.

That's even better, I wasn't aware of that. Looking at [Apple's website](https://developer.apple.com/support/compare-memberships/) it looks like it's $99 to have a team membership too.

>I don't think I ever proposed crowd funding FWIW. I have an account I pay for every year anyway, and I'm a Vimari user and I want it to work in Safari 12, so I'm fine just publishing the extension under my account. 

Sorry for putting words in your mouth, my apologies. That's how I interpreted it.

>I understand the other legitimate concerns about ownership, but I just want to be clear that I'm willing to do this on my own without collecting money from anyone.

I really appreciate that, and I don't want to collect money from people just for the fun of it but sharing this responsibility would feel better. And it would also take the potential burden off of anyone owning this extension in the gallery. Being responsible on your own for these things can be stressful, and even though this project does not get a lot of bug reports or feature requests right now I think it will attract more attention once we're unblocked from releasing new versions.

>Another thing to consider is that if people contribute money, what, if any, obligations should be upheld in return. For example, one could imagine a commitment to keep the extension published and updated to support new operating system versions over the course of a year at a minimum. Maybe critical bug fixes too. Then you'd have to think about if it also includes any kind of active feature development. Whatever the decision is, that should be made clear when asking for monetary contributions.

A very good point, I was just thinking about this just before writing this actually. [Open Collective](https://opencollective.com/learn-more) has useful information about this and examples on how other projects handle this. I completely agree with you. Asking for money for this should be handled carefully and with full transparency.

**Comment by simeg** [comment]
What I'm looking for in this issue are thoughts on starting a funding campaign for a Apple team subscription. I'm willing to write up a summary of what it would mean to get us funded, in terms of commitment towards this extension. I like what @jad mentioned above about committing to fixing critical bugs etc. I am committing to working on this extension for the upcoming year, if we are able to release new versions.

**Use this post to vote for if you think we should start a crowd funding campaign**. If you want to explain your vote in a comment then please do. Thank you.

**Comment by jaredmichaelwilliams** [comment]
ill contribute $20,

also i have a developer subscription already if you want 

as for packaging it with the app store you can most likely use fastlane to do it easy, and quickly 

https://fastlane.tools

**Comment by simeg** [comment]
Thanks for voting everyone, when writing this there's 8 👍's and to me that's enough to start this process. I've put together a document where I specify everything a potential donator would like to know, and I've setup a donation button. Please read it. If you feel like something is missing or something is wrong, let me know.

**Comment by simeg** [comment]
I'm going on vacation out of the country for a few days now, so I won't be able to update the README with the new status. Will update asap when I'm back!

**Comment by simeg** [comment]
We are funded, people! It's so cool to see people coming together for this. I had to add a few dollars to the goal because of unforeseen PayPal fees, I think I added maybe $7 to the total.

I want to say thank you but I'm not sure if that's the right word to use, because it's like we're all helping each other. I can say though that people have been very generous, and that's so cool.

I just used the money to enroll into the Apple Developer Program, and I will start looking into how to put the extension on the Safari Extension store as soon as possible. I'll create a new issue to keep you all updated on the latest news. I feel like that is part of the crowdfunding process - to be transparent about what I'm doing with the money you donated.
